packages/@aws-cdk/core/lib/custom-resource-provider/custom-resource-provider.ts

```diff
@@ -200,16 +201,17 @@ export class CustomResourceProvider extends Construct {
 
     const stack = Stack.of(scope);
 
-    // copy the entry point to the code directory
-    fs.copyFileSync(ENTRYPOINT_NODEJS_SOURCE, path.join(props.codeDirectory, `${ENTRYPOINT_FILENAME}.js`));
-
     // verify we have an index file there
     if (!fs.existsSync(path.join(props.codeDirectory, 'index.js'))) {
       throw new Error(`cannot find ${props.codeDirectory}/index.js`);
     }
 
+    const stagingDirectory = FileSystem.mkdtemp('cdk-custom-resource');
+    fs.copyFileSync(path.join(props.codeDirectory, 'index.js'), path.join(stagingDirectory, 'index.js'));
```

Comment:
Previously we were copying  the `ENTRYPOINT_FILENAME` into the `codeDirectory`. Now we are copy the `ENTRYPOINT_FILENAME` and _just_ the `index.js` file from the `codeDirectory` into a tmp directory. What happens if some custom resources have more than just an `index.js` file that needs to be copied?

For example https://github.com/aws/aws-cdk/tree/main/packages/%40aws-cdk/aws-iam/lib/oidc-provider

Reply:
You're absolutely right! Done.